java-checks-test-sources/src/main/java/checks/VarCanBeUsedCheck.java

```diff
@@ -66,9 +66,14 @@ void f(int[] array) {
     char c = i == 3 ? 0 : "hdr".charAt(i); // Compliant, despite it's possible here, ternary  operator may bring confusion in understanding the inferred type
 
     byte[] input = new byte[array.length]; // Noncompliant
-    
     var input2 = new byte[array.length]; // Compliant
 
+    String[] arrayInit = new String[]{"A", "B", "C"}; // Noncompliant
+    var arrayInit2 = new String[]{"A", "B", "C"}; // Compliant
+
+    String[] arrayInit3 = {"A", "B", "C"}; // Compliant, for array initializer, without type, it is not possible to change to var
+    //var arrayInit4 = (String[]) {"A", "B", "C"}; // does not compile
```

Comment:
Ift this line does not compile, we should consider moving it to a non-compiling file or deleting it altogether
```suggestion
```